pkg/otlp/collector.go

```diff
@@ -63,6 +65,7 @@ func getComponents(s serializer.MetricSerializer) (
 		otlpexporter.NewFactory(),
 		serializerexporter.NewFactory(s),
 		loggingexporter.NewFactory(),
+		logsagentexporter.NewFactory(logsAgentChannel),
```

Comment:
I'd suggest to do the followings instead:
1. Add a func `GetRunningAgent` in pkg/logs
2. Call `GetRunningAgent` here only when `otlp_config::logs` is enabled, then get the `logsAgentChannel` from logs agent
3. Revert all changes to the public func in this file